### ISSUE TITLE
WordPress.com on Atomic / Jetpack: Create HE tool to reconnect Jetpack to WPCOM Atomic sites.

### ISSUE THREAD

**Issue by JoshuaGoode** [Automattic/wp-calypso#57602]
### What

An internal HE tool to allow for easier WordPress.com on Atomic Jetpack reconnections

### Why

Reconnecting Jetpack on WordPress.com on Atomic sites remains a top time-consuming pain point for users and our support teams.

Internal overviews:
p58i-a2c-p2 - extended pain points review.
P9HQHe-iZ-p2 - reconnection process.
p9F6qB-7xg-p2 - newer issues. 

Work was done to help improve Jetpack connections, auto-heal them, and make them more resilient (see internal comment thread on p58i-a2c-p2 ).

However, there are still many situations and interactions where sites need a manual Jetpack to reconnect performed by an HE and this will remain the case until WPCOM Atomic Jetpack connections can fix themselves.

### How

Explore utilizing `wp jetpack authorize_user` and/or other options to provide an internal HE reconnection tool. 

That command does require that a token already exists and it must be pulled from a specific WPCOM table.  Depending on the connection and issue, a blog token might need to be set as well.

Specifically, this would be used to reconnect a site owner to a site that they were already connected to but something broke the connection.

Quick internal discussion regarding `jetpack auhtorize_user`:

p1635947078022200-slack-CDLH4C1UZ

---

This will likely be difficult but it will be worth exploring potential options with Jetpack folks.

**Comment by Robertght** [comment]
@Automattic/happops-devs is this something that you could work on?

**Comment by klimeryk** [comment]
I'm happy to help make sure the tool is well integrated in our support flows, but the actual implementation should probably be done by a product team that knows the details, most likely the Atomic Team. @jmdodd, is this on your radar?

**Comment by JoshuaGoode** [comment]
> I'm happy to help make sure the tool is well integrated in our support flows, but the actual implementation should probably be done by a product team that knows the details, most likely the Atomic Team. @jmdodd, is this on your radar?

This would generally be outside the Atomic platform team's scope and more appropriate for a WPCOM-specific product team.

**Comment by klimeryk** [comment]
Thanks for the reply, @jmdodd 👍 Just so that I understand the context here better: is this one of those issues that are not possible to solve permanently? In other words, the design of the process/system sometimes necessitates reconnecting it manually like this? If that's the case and we do indeed a tool for it for the foreseeable future, I'll cc @davemart-in to help figure out who would be best on WPCOM's side to tackle this. Ideally, it'd _not_ be an HE only tool, but a user-facing tool first.

**Comment by jmdodd** [comment]
The reconnection process can require decisions that are not easily encoded for programmatic solution: matching usernames but different user IDs and email addresses, matching different emails but similar usernames, multiple usernames with numbers appended as the customer created test accounts on their WoA site, nothing matches but the customer has informed us that they switched from X.com to Y.com for email, etc. /cc @fgiannar as I know she has done a lot of work around auto-healing and tokens.

The Atomic team can provide guidance on what features can be used to implement tools for reconnection Jetpack, but because every site is different there will be outliers that require assistance. We can't allow an arbitrary customer to demand that someone else's account be connected; we cannot trust the remote site's information beyond what we store in persistent (non-customer-alterable) data and to a more limited extent what is stored in the remote database.

**Comment by fgiannar** [comment]
> However, there are still many situations and interactions where sites need a manual Jetpack to reconnect performed by an HE and this will remain the case until WPCOM Atomic Jetpack connections can fix themselves.

Atomic Jetpack **user** connections are already healing themselves when the following conditions are met:
- An existing user token on WPCOM
- The remote user email matches the WPCOM user email

p1HpG7-bKn-p2 - Self-healing user connections on WoA sites

The most popular cases where self-healing is **not** successful are:
- The domain is not pointing to our servers, aka it's not actually a WoA site 
- The remote user email doesn't match the WPCOM user email. This means that we cannot know for sure which user to reconnect
<img width="1241" alt="Screenshot 2021-11-10 at 1 09 14 PM" src="https://user-images.githubusercontent.com/1758399/141102738-96674579-861d-4a02-ab28-4dbb83b84c66.png">

Both the above cases result in flagging them in Blog RC as **needing manual intervention**.

>  Ideally, it'd not be an HE only tool, but a user-facing tool first.

Not sure if we need a user-facing tool, given we already display a corresponding alert in such cases which prompts the user to reconnect. 

> Explore utilizing wp jetpack authorize_user and/or other options to provide an internal HE reconnection tool.

One possibility that comes to mind regarding a reconnection tool is to utilize the existing self-healing process but that would require an HE **manually** solving the case that **prevents the automatic** self-healing in the first place. That would be either the domain issue or updating the wpcom user email to match the remote one.

In any case, the most important aspect here would be **verifying the user's intent to actually connect/reconnect** their WPCOM account. This tool should **NEVER** be used without getting permission from the user first.

If we decide to move forward I'd be happy to provide further guidance if needed.

**Comment by JoshuaGoode** [comment]
> Atomic Jetpack user connections are already healing themselves when the following conditions are met:

The auto-healing work has helped significantly, however, HEs still face hundreds of manual Jetpack reconnections a month.  For the last 30 days, we saw at least 200 instances and it was likely closer to 350+.  

A portion of these require no email changes, a connection did previously exist,  and they require reconnecting the site and user manually. 

Often, these are in live chat so we cannot wait for an auto-heal attempt.  The HE must then proceed with the reconnect steps on P9HQHe-iZ-p2.  

I've moved this discussion to a P2: p9F6qB-7G3-p2
